### ISSUE TITLE
Exclusion regex passed to match-dir does not exclude

### ISSUE THREAD

**Issue by tomarak** [PyCQA/pydocstyle#175]
When I supply a regex to the `match-dir` option to ignore my migrations folders, the migrations folders still get included in the directories checked. I've tried:

```
match-dir="[^(\.|migrations)].*”
```

```
match-dir="^((?!migrations.)*$”
```

```
match-dir="^(?!migrations).*”
```

And none work as expected.

The file structure of my project looks like this:

```
project/
│
└───app1/
|    │
|    ├───migrations/
│
└───app2/
|    │
|    ├───migrations/
│
└───app3/
     │
     ├───migrations/

```

**Comment by goanpeca** [comment]
Having an exclude option via cli and config (like pep8, flake8 and a lot of other tools) would make this easier

**Comment by mheppner** [comment]
An alternative solution to ignore the migrations themselves (files starting with 4 digits). This isn't ideal, but it helps:

```match = (?!test_)(?!\d{4}_).*\.py```

**Comment by cdeil** [comment]
+1 to add something simple like `--exclude` from `flake8`.
Or alternatively, to add a docs section that shows working examples how to ignore given files or folders. (this issue suggests that `match-dir` option processing is broken at the moment. Is this correct?)

**Comment by mrshannon** [comment]
I agree this is a very confusing method to include/exclude directories and a clearer method, such as that in `pycodestyle` is needed.  However, the last regex
```
match-dir="^(?!migrations).*”
```
does work in version 2.3.1, not sure at which point this was fixed.  The first two regexes are not valid for the OP's purpose.  The first will skip any directories beginning with '.', 'm', 'i', 'g', 'r', 'a', 't', 'i', 'o'. 'n', or 's' while the second only matches the empty string.

Therefore, this issue should probably be closed.

**Comment by craiga** [comment]
Still appears to be an issue in 2.1.1. `"^(?!migrations).*”` appears to be excluding all directories in my project.

**Comment by dalepotter** [comment]
> does work in version 2.3.1

Is this even a version? Not according to https://github.com/PyCQA/pydocstyle/tags

**Comment by StrikerRUS** [comment]
`2.1.1` with explicit default `--match-dir='[^\.].*'` ignores all folders.

**Comment by StrikerRUS** [comment]
Docs are lying: this option works **without** quotes, like: `"^(?!migrations).*`

**Comment by shacharoo** [comment]
@tomarak and @craiga - your issue is that your closing double-quote isn't ASCII, it's some Unicode character, probably copied from somewhere:
![image](https://user-images.githubusercontent.com/13560060/46913550-c06a3f80-cf97-11e8-83e9-5046272daf6a.png)

Other than that, there is a difference between how different shells parse singe and double quotes, so "which quotes should I use" is completely a user decision based on his/hers setup. Take a look at [this SOVF question](https://stackoverflow.com/questions/6697753/difference-between-single-and-double-quotes-in-bash) for more info.

Bottom line: there is no issue with how `pydocstyle` behaves, but I agree that we can add some warning in the documentation about this, which I will do today 😺 

Cheers,
Shachar.

**Comment by craiga** [comment]
Well, that's a little embarrassing. Thanks for your patience.

**Comment by Malachov** [comment]
Finally found solution - at least for VS Code (but don't ask me why it works though :)
```json
"python.linting.pydocstyleArgs": ["--match=^((?!test).)*$"],
```
